### ISSUE TITLE
Can't open solutions in Visual Studio (2017)

### ISSUE THREAD

**Issue by RayKoopa** [RedpointGames/Protogame#157]
I'm having troubles getting started with Protogame. I'm on Windows 10, using VS2017 Community (fresh OS and tools installation from 14th April).

Whenever I created a new project and want to load it in VS2017, I get stuck at `Preparing solution` and get `Protoinject.FactoryGenerator has stopped working`. Trying to debug the error shows me it tried to load non-existant assemblies in `PreloadedAssemblyResolver.cs`:

    System.IO.FileNotFoundException: 'Could not find file 
    'D:\Projects\Protogame\TestGame\Protogame\ProtogamePostLibBuild\bin\Windows\AnyCPU\Debug\ProtogamePostLibBuild.exe'.

The Debug folder exists, but assemblies have never been dropped into them, nothing seems to have been built for the included Protogame solution.

Trying to open *that* solution (`..\Protogame\Protogame.Windows.sln`) fails aswell, as it starts the Protoinject.FactoryGenerator again who then fails with

    System.IO.FileNotFoundException: 'Could not find file
    'D:\Projects\Protogame\TestGame\Protogame\Protogame\bin\Windows\AnyCPU\Debug\Protogame.dll'.

Did I do something wrong?

**Comment by hach-que** [comment]
I remember seeing this issue occur for someone else, and it turned out to be something weird like having scripts set for CMD.exe which change the working directory (maybe something like this: http://stackoverflow.com/questions/17404165/how-to-run-a-command-on-command-prompt-startup-in-windows?)

From the command line, when you're in the directory where your game is located, try running `Protobuild.exe --generate` then `Protobuild.exe --build` and copy-paste the output here if possible.

**Comment by RayKoopa** [comment]
Ah yeah, I remember using that key once, but on this system, there are no global commands set up to run when starting CMD.

The commands do not cause any errors (build just a lot of warnings), and after that I can open the solution. Pretty weird, as if something's set up differently when you issue those commands from inside of VS?

PS: Another "anomaly" I noticed is that when the project was generated, the docs say VS is started automatically, but I had to open it manually (nothing happened after creation was complete).

**Comment by hach-que** [comment]
Yeah, VS2017 doesn't autostart from Protobuild because they removed all the registry keys that tell applications where Visual Studio is installed.  They have a C# library for it, but since Protobuild Manager is technically a single executable app, I can't just add a dependency to it like that.  So for now, you need to open VS2017 solutions manually.

Can you run Protobuild from the command-line and paste the output?  Without that I can't really diagnose further.

**Comment by RayKoopa** [comment]
I've uploaded the output of `--generate` and `--build` here (since it's quite a bit): http://ray.syroot.com/protogame/protobuild_output.txt

**Comment by hach-que** [comment]
Oh hmm.  Since now that it's been built on the command line, it works in Visual Studio, you can hopefully replicate the original scenario by deleting the Protogame folder in your project, running `Protobuild.exe --generate` (but not build), [setting the build output in Visual Studio to Detailed](http://sedodream.com/2008/06/16/ChangingBuildVerbosityInVisualStudio.aspx), the running the build from Visual Studio and copy-pasting the content of the Build Output window.

**Comment by hach-que** [comment]
My current theory is that Visual Studio is building the projects in the wrong order (or without considering dependencies).  I've noticed this happens on rare occasion; normally it's fixable by just running the build again in Visual Studio.

**Comment by RayKoopa** [comment]
I created a new TestGame project from scratch, ran `--generate` from the command line, then opened it in VS2017 and built from there (output set to Detailed), but it was quite unspectacular and worked fine: http://ray.syroot.com/protogame/protobuild_output2.txt

EDIT: What is even more weird now: I created yet another project, and then opened the solution in VS directly - I no longer get the `Protoinject.FactoryGenerator has stopped working` error which I always got yesterday when it's preparing the solution... 😕 Guess it's working fine now...

**Comment by hach-que** [comment]
My guess is you ran into the scenario I described above; where Visual Studio occasionally builds projects in the wrong order, despite the dependencies that are specified.  I haven't been able to track down what causes it (especially since it occurs so intermittently).

That said, the factory generator probably should give a better error message than "stopped working".  We should probably output something like "Input file was not available at build time; try building the solution again to resolve the issue."

**Comment by RayKoopa** [comment]
Yeah, it should at least not crash because then you're stuck with "Preparing solution" and can't open it in VS at all to build it :o
